.github/workflows/deploy-production.yml

```
@@ -1,62 +1,308 @@
 
-      - name: Pre-deployment validation
+      - name: Generate deployment metadata
+        id: deployment
+        run: |
+          DEPLOYMENT_ID="deploy-$(date +%Y%m%d-%H%M%S)-${{ github.sha::8 }}"
+          echo "deployment_id=$DEPLOYMENT_ID" >> $GITHUB_OUTPUT
+          echo "::notice title=Deployment ID::$DEPLOYMENT_ID"
+
+      - name: Generate image tag
+        id: image
+        run: |
```

actionlint (1.7.7)

got unexpected character ':' while lexing expression, expecting 'a'..'z', 'A'..'Z', '_', '0'..'9', ''', '}', '(', ')', '[', ']', '.', '!', '<', '>', '=', '&', '|', '*', ',', ' '

(expression)